drivers/pds/lib/utils.go

```diff
@@ -1715,18 +1790,21 @@ func GetAllSupportedDataServices() map[string]string {
 func UpdateDataServices(deploymentID string, appConfigID string, imageID string, nodeCount int32, resourceTemplateID, namespace string) (*pds.ModelsDeployment, error) {
 
 	log.Infof("depID %v appConfID %v imageID %v nodeCount %v resourceTemplateID %v", deploymentID, appConfigID, imageID, nodeCount, resourceTemplateID)
-	deployment, err = components.DataServiceDeployment.UpdateDeployment(deploymentID, appConfigID, imageID, nodeCount, resourceTemplateID, nil)
-	if err != nil {
-		log.Errorf("An Error Occured while updating deployment %v", err)
-		return nil, err
-	}
+	err = wait.Poll(maxtimeInterval, timeOut, func() (bool, error) {
+		deployment, err = components.DataServiceDeployment.UpdateDeployment(deploymentID, appConfigID, imageID, nodeCount, resourceTemplateID, nil)
+		if err != nil {
+			log.Errorf("An Error Occured while updating deployment %v", err)
+			return false, nil
```

Comment:
Don't log the error just return it if you need to, and don't you want to return this error? Why are we returning `nil` here?

Reply:
Removed the error logs